sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1101,6 +1082,91 @@ def slow_haversine_distances(x, y):
         haversine_distances(X)
 
 
+@pytest.mark.parametrize(
+    "metric_fn", [nan_euclidean_distances, nan_manhattan_distances]
+)
+@pytest.mark.parametrize("missing_value", [np.nan, -1])
+def test_nan_distances_metrics_complete_nan(metric_fn, missing_value):
+    X = np.array([[missing_value, missing_value], [0, 1]])
+    exp_dist = np.array([[np.nan, np.nan], [np.nan, 0]])
+
+    dist = metric_fn(X, missing_values=missing_value)
+    assert_allclose(exp_dist, dist)
+
+    dist = metric_fn(X, X.copy(), missing_values=missing_value)
+    assert_allclose(exp_dist, dist)
+
+
+@pytest.mark.parametrize(
+    "metric_fn", [nan_euclidean_distances, nan_manhattan_distances]
+)
+@pytest.mark.parametrize("X", [np.array([[np.inf, 0]]), np.array([[0, -np.inf]])])
+@pytest.mark.parametrize("Y", [np.array([[np.inf, 0]]), np.array([[0, -np.inf]]), None])
+def test_nan_euclidean_distances_infinite_values(metric_fn, X, Y):
+    with pytest.raises(ValueError) as excinfo:
+        metric_fn(X, Y=Y)
+
+    exp_msg = "Input contains infinity or a value too large for dtype('float64')."
+    assert exp_msg == str(excinfo.value)
+
+
+def test_nan_manhattan_distances_equal_to_manhattan_distances():
+    rng = np.random.RandomState(714)
+    X = rng.randn(3, 4)
+    Y = rng.randn(4, 4)
+    normal_distance = manhattan_distances(X, Y=Y)
+    nan_distance = nan_manhattan_distances(X, Y=Y)
+    assert_allclose(normal_distance, nan_distance)
+
+
+@pytest.mark.parametrize("missing_value", [np.nan, -1])
+def test_nan_manhattan_distances_not_trival(missing_value):
```

Comment:
In this test, I think `assert_array_equal` should be preferred over `assert_allclose`.